### PR TITLE
fix: incorrect python version was used in some parts of the solve

### PR DESCRIPTION
**Pull request by tdejager**
This fixes the over-use of the `Intepreter` in the solve, it was using it incorrectly in some parts of the resolution. This would result in weird behavior, like getting the incorrect wheels from the registry and not being to able to match them with the wheel tags, for example.

Discovered because of: https://github.com/prefix-dev/pixi/issues/771#issuecomment-2476195579. This should now work.

## Test

To test you can try:

```
[project]
channels = ["conda-forge"]
name = "test"
platforms = ["osx-arm64", "linux-64"]

[dependencies]
scipy = ">=0.19"

[pypi-dependencies]
finch-mlir = ">=0.0.2"
```

Which did not work before and now it does.

### CODE REVIEW

**Review by ruben-arts — APPROVED** [review]
Appart from the warning, this looks good to me! 

We should quickly get this in a release.

File: src/lock_file/update.rs [hunk 1]
```diff
@@ -1943,6 +1943,7 @@ async fn spawn_solve_pypi_task(
                     "missing python interpreter from environment"
                 )
             })?;
+        tracing::warn!("{:?}", &python_path);
```
[the remark this answers is not in the document]
Comment (tdejager) [reply src/lock_file/update.rs:1946]: Yeah thanks!

File: src/lock_file/resolve/pypi.rs [hunk 2]
```diff
@@ -212,12 +213,38 @@ pub async fn resolve_pypi(
     // Determine the tags for this particular solve.
     let tags = get_pypi_tags(platform, &system_requirements, python_record.as_ref())?;
 
-    // Construct an interpreter from the conda environment.
+    // We need to setup both an interpreter and a requires_python specifier.
+    // The interpreter is used to (potentially) build the wheel, and the requires_python specifier is used
+    // to determine the python version of the wheel.
+    // So make sure the interpreter does not touch the solve partgs of this function
```
Comment (hameerabbasi) [inline src/lock_file/resolve/pypi.rs:219]:
```suggestion
    // So make sure the interpreter does not touch the solve parts of this function
```

**Review by hameerabbasi — CHANGES_REQUESTED** [review]
Typo, plus maybe we should add tests?

**Pull request merged**